backend/ts/src/main/scala/aqua/backend/ts/TypescriptFunc.scala

```diff
@@ -18,16 +18,25 @@ case class TypescriptFunc(func: FuncCallable) {
 
     val tsAir = FuncAirGen(func).generateClientAir(conf)
 
-    val returnCallback = func.ret.as {
-      s"""h.onEvent('${conf.callbackService}', '${conf.respFuncName}', (args) => {
-         |  const [res] = args;
-         |  resolve(res);
-         |});
-         |""".stripMargin
+    val returnCallback = func.ret.map(_._2).map {
+      case OptionType(_) =>
+        s"""h.onEvent('${conf.callbackService}', '${conf.respFuncName}', (args) => {
+           |  const [res] = args;
+           |  resolve(res.length === 0 ? null : res[0]);
```

Comment:
here `res` could be undefined as I see

Reply:
How? It is always passed from AIR, as it was before, and has an array type.